homeassistant/components/switchbee/climate.py

```diff
@@ -0,0 +1,206 @@
+
+FAN_SB_TO_HASS = {
+    ThermostatFanSpeed.AUTO: FAN_AUTO,
+    ThermostatFanSpeed.LOW: FAN_LOW,
+    ThermostatFanSpeed.MEDIUM: FAN_MEDIUM,
+    ThermostatFanSpeed.HIGH: FAN_HIGH,
+}
+
+FAN_HASS_TO_SB = {
+    FAN_AUTO: ThermostatFanSpeed.AUTO,
+    FAN_LOW: ThermostatFanSpeed.LOW,
+    FAN_MEDIUM: ThermostatFanSpeed.MEDIUM,
+    FAN_HIGH: ThermostatFanSpeed.HIGH,
+}
+
+HVAC_MODE_SB_TO_HASS = {
+    ThermostatMode.COOL: HVACMode.COOL,
+    ThermostatMode.HEAT: HVACMode.HEAT,
+    ThermostatMode.FAN: HVACMode.FAN_ONLY,
+}
+
+HVAC_MODE_HASS_TO_SB = {
+    HVACMode.COOL: ThermostatMode.COOL,
+    HVACMode.HEAT: ThermostatMode.HEAT,
+    HVACMode.FAN_ONLY: ThermostatMode.FAN,
+}
+
+HVAC_ACTION_SB_TO_HASS = {
+    ThermostatMode.COOL: HVACAction.COOLING,
+    ThermostatMode.HEAT: HVACAction.HEATING,
+    ThermostatMode.FAN: HVACAction.FAN,
+}
+
+HVAC_UNIT_SB_TO_HASS = {
+    ThermostatTemperatureUnit.CELSIUS: TEMP_CELSIUS,
+    ThermostatTemperatureUnit.FAHRENHEIT: TEMP_FAHRENHEIT,
+}
+
+SUPPORTED_FAN_MODES = [FAN_AUTO, FAN_HIGH, FAN_MEDIUM, FAN_LOW]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbee thermostat."""
+    coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeClimate(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if isinstance(switchbee_device, SwitchBeeThermostat)
+    )
+
+
+class SwitchBeeClimate(SwitchBeeDeviceEntity[SwitchBeeThermostat], ClimateEntity):
+    """Representation of an Switchbee button."""
+
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+    )
+    _attr_fan_modes = SUPPORTED_FAN_MODES
+    _attr_target_temperature_step = 1
+
+    def __init__(
+        self,
+        device: SwitchBeeThermostat,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the Switchbee switch."""
+        super().__init__(device, coordinator)
+        # set HVAC capabilities
+        self._attr_max_temp = device.max_temperature
+        self._attr_min_temp = device.min_temperature
+        self._attr_temperature_unit = HVAC_UNIT_SB_TO_HASS[device.unit]
+        self._attr_hvac_modes = [HVAC_MODE_SB_TO_HASS[mode] for mode in device.modes]
+        self._attr_hvac_modes.append(HVACMode.OFF)
+        self._update_device_attrs(self._device)
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        return self._attr_hvac_mode
+
+    @property
+    def fan_mode(self) -> str:
+        """Return the fan setting.
+
+        Requires ClimateEntityFeature.FAN_MODE.
+        """
+        return self._attr_fan_mode
+
+    @property
+    def target_temperature(self) -> int:
+        """Return the temperature we try to reach."""
+        return self._attr_target_temperature
+
```

Comment:
These are already defined in the base class, you don't need to override them:
https://github.com/jafar-atili/core/blob/50c278424ebaaa8964aa822137d2c289ee39cdaa/homeassistant/components/climate/__init__.py#L384
```suggestion
```

Reply:
I know but I override them with specific return value as they were failing `mypy` as they returns `Optional[<Type>]`

Reply:
It is better to add `assert ... is not None` to help mypy in those cases.